### PR TITLE
Feature/update books

### PR DESCRIPTION
**Pull request by AlexAlt**
Adds an update Controller function to handlle updating as many fields as provided.

### CODE REVIEW

File: test/objectUtils_spec.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  describe('compact', function() {
+    it('should remove undefined values from an object', function() {
+      const obj = {
+        'title': 'Book 1',
+        'author': undefined,
+      };
+
+      const compacted = compact(obj);
+      expect(compacted).to.not.have.property('author');
+      expect(compacted).to.not.have.property('meta');
```
Comment (AlexAlt) [inline test/objectUtils_spec.js]:
```suggestion
      
```

File: test/objectUtils_spec.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  describe('compact', function() {
+    it('should remove undefined values from an object', function() {
+      const obj = {
+        'title': 'Book 1',
+        'author': undefined,
+      };
+
+      const compacted = compact(obj);
+      expect(compacted).to.not.have.property('author');
+
```
Comment (AlexAlt) [inline test/objectUtils_spec.js]:
```suggestion
```

File: model/Book.js [hunk 3]
```diff
@@ -11,6 +11,14 @@ const bookSchema = new Schema({
   }
 });
 
+
+bookSchema.methods.updateAndSaveMultipleFields = function(compactedParams = {}) {
+  Object.entries(compactedParams).forEach(param => {
+    this[`${param[0]}`] = param[1]
+  });
```
Comment (DTwigs) [inline model/Book.js:18]: would something like this work instead:
```
this = {
  ...this,
  ...params
};
```

File: model/Book.js [hunk 4]
```diff
@@ -11,6 +11,14 @@ const bookSchema = new Schema({
   }
 });
 
+
+bookSchema.methods.updateAndSaveMultipleFields = function(compactedParams = {}) {
```
Comment (DTwigs) [inline model/Book.js]: id rename the argument to just params.

File: model/Book.js (hunk 3 above)
Comment (DTwigs) [inline model/Book.js]:
```
for (const key in params) {
  if(params.hasOwnProperty(key)) {
    this[key]=params[key];
  }
}
```

File: views/books.html [hunk 5]
```diff
@@ -50,9 +56,37 @@ <h3>Completed</h3>
...
+          body: JSON.stringify({ readOn: new Date().toISOString(), currentlyReading: false })
+        })
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error(response.message);
+          }
+        })
+        .then(() => {
+          window.location.reload();
+        })
```
Comment (DTwigs) [inline views/books.html]: this will reload even if theres an error i believe.

**Pull request merged**